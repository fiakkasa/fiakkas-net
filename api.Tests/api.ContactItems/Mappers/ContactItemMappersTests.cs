using api.ContactItems.Interfaces;
using api.ContactItems.Models;

namespace api.ContactItems.Mappers.Tests;

public class ContactItemMappersTests
{
    public record ContactItemMockEntity : IContactItem
    {
        public Guid Id { get; init; }
        public DateTimeOffset CreatedAt { get; init; }
        public DateTimeOffset? UpdatedAt { get; init; }
        public long Version { get; init; }
        public string Key { get; init; } = string.Empty;
        public string? Icon { get; init; }
        public string Title { get; init; } = string.Empty;
        public string Description { get; init; } = string.Empty;
        public Uri? Href { get; init; }
    }

    [Fact]
    public void Map_Should_Return_Data()
    {
        var item = new ContactItemMockEntity
        {
            Id = new Guid("ebf224a8-7ff3-47b9-882b-dd41ec7f5a05"),
            CreatedAt = new(2024, 1, 1, 0, 0, 0, TimeSpan.Zero),
            UpdatedAt = null,
            Version = 1,
            Key = "Key",
            Icon = "Icon",
            Title = "Title",
            Description = "Content",
            Href = new Uri("/test", UriKind.Relative)
        };

        var result = item.Map();

        result.Should().BeOfType<ContactItem>();
        result.MatchSnapshot();
    }
}
