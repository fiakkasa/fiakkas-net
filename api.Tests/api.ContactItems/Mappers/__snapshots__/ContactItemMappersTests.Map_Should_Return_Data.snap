{
  "Key": "Key",
  "Icon": "Icon",
  "Title": "Title",
  "Content": "Content",
  "Href": "/test",
  "Id": "ebf224a8-7ff3-47b9-882b-dd41ec7f5a05",
  "CreatedAt": "2024-01-01T00:00:00+00:00",
  "UpdatedAt": null,
  "Version": 1
}
